modules/widgets/social-icons.php

```diff
@@ -512,6 +512,106 @@ public function get_supported_icons() {
 				'icon'  => 'feed',
 				'label' => __( 'RSS Feed', 'jetpack' ),
 			),
+			array(
```

Comment:
It's a bit annoying that we have so many entries for just one thing: RSS. I wonder if there would be a way to shorten this a bit by having one big array of potential RSS formats that we would then add all at once to this existing list?

Reply:
Yes, it is a bit unwieldy to have so many! Do you mean put all the RSS parameters into an array in the 'url', e.g:

`array(
    'url'   => array( '/feed/', '/feeds/', 'format=RSS', '/rss' ),
    'icon'  => 'feed',
    'label' => __( 'RSS Feed', 'jetpack' ),
)`

If so then it would require changing the way the urls are matched when the widget is rendered, currently it uses `if ( false !== stripos( $icon['url'], $social_icon['url'] ) ) {` within a foreach loop to see if any of the social_icon urls match the URL to be rendered, so I think we'd have to check to see if the `$icon['url']` was an array or not, then match inside that array if it is. Does that seem a viable way?

Reply:
Yes, you could even pull that array of feed URL formats out of there, in a new `$feed_url_formats` array, and only add it later to the `$social_links_icons` array, right before it's returned, all inside
`get_supported_icons()`.

Reply:
Hi, I've pushed some changes along those lines now. I added a new `$feed_url_formats` array to hold the 'feed' url formats. I've then updated the code that renders the icons to check if the 'url' is an array and if so then to loop over the array and match inside that, if not an array then it matches as normal. 

This does mean there are two bits of identical code echoing the icon out which I'm not too happy about, but in the interest of not over-optimising the code I've gone with that for now. Happy for feedback / suggestions on better ways that could be handled if you think that needs more attention.